apps/dashboard/src/contract-ui/hooks/useRouteConfig.tsx

```diff
@@ -121,6 +124,70 @@ export function useContractRouteConfig(
       "ERC20ClaimPhasesV2",
     ],
   });
+
+  // ContractSettingsPage
+  const detectedMetadata = extensionDetectedState({
+    contractQuery,
+    feature: "ContractMetadata",
+  });
+  const detectedPrimarySale = extensionDetectedState({
+    contractQuery,
+    feature: "PrimarySale",
+  });
+  const detectedRoyalties = extensionDetectedState({
+    contractQuery,
+    feature: "Royalty",
+  });
+  const detectedPlatformFees = extensionDetectedState({
+    contractQuery,
+    feature: "PlatformFee",
+  });
+
+  // ContractTokensPage
+  const isERC20 = detectFeatures(contractQuery.contract, ["ERC20"]);
+
+  const isERC20Mintable = detectFeatures(contractQuery.contract, [
+    "ERC20Mintable",
+  ]);
+
+  const isERC20Claimable = detectFeatures(contractQuery.contract, [
+    "ERC20ClaimConditionsV1",
+    "ERC20ClaimConditionsV2",
+    "ERC20ClaimPhasesV1",
+    "ERC20ClaimPhasesV2",
+  ]);
+
+  // AccountPage
+  const detectedAccountFeature = extensionDetectedState({
+    contractQuery,
+    feature: ["Account"],
+  });
+
+  // ContractEmbedPage
+  const { erc20, erc1155, erc721 } = getErcs(contractQuery?.contract);
+  const isMarketplaceV3 = detectFeatures(contractQuery?.contract, [
+    "DirectListings",
+    "EnglishAuctions",
+  ]);
+  const ercOrMarketplace =
+    contractTypeQuery.data === "marketplace"
+      ? "marketplace"
+      : isMarketplaceV3
+        ? "marketplace-v3"
+        : erc20
+          ? "erc20"
+          : erc1155
+            ? "erc1155"
+            : erc721
+              ? "erc721"
+              : null;
+
+  // AccountPermissionsPage
+  const detectedPermissionFeature = extensionDetectedState({
+    contractQuery,
+    feature: ["AccountPermissions", "AccountPermissionsV1"],
+  });
```

Comment:
I THINK the only dynamic thing these depend on is the contractQuery -> aka can u move all of these into a `useMemo()` with the only dependency then being the contractQuery.contract? If so that would likely be a good idea!

Reply:
So this will be like a huge object containing the data, wrapped inside a useMemo